ckanext/s3filestore/tests/test_uploader.py

```diff
@@ -217,3 +218,46 @@ def test_resource_upload_with_url_and_clear(self):
                        'id': '',    # Empty id from the form
                        'url': 'http://asdf', 'save': 'save'},
                  extra_environ=env)
+
+    def test_is_presigned_url(self):
+        ''' Tests that presigned URLs are correctly recognised.'''
+        assert_true(_is_presigned_url('https://example.s3.amazonaws.com/resources/foo?AWSAccessKeyId=SomeKey&Expires=9999999999Signature=hb7%2F%2Bz1H%2B8wdEy0pCsX7bZG%2BuPU%3D'))
+        assert_false(_is_presigned_url('https://example.s3.amazonaws.com/resources/foo'))
+
+    @helpers.change_config('ckanext.s3filestore.acl', 'auto')
+    def test_resource_url(self):
```

Comment:
please split this into at least 3 tests
test_resource_url_should_be_public_for_publicDataset
test_resource_url_should_be_private_for_privateDataset
test_resource_url_should_update_state_if_private_and_dataset_made_public
test_resource_url_should-update_state_if_public_and_dataset_made_private

also if not already exisitng
test file delete is called

unsure if we should cover this edge case, if filename is renamed, delete old file when uploading new file (after successful upload? )

Reply:
I can do this, but `test_resource_url_should-update_state_if_public_and_dataset_made_private` seems like it will necessarily be a superset of `test_resource_url_should_be_public_for_publicDataset` (since it will first create a public dataset and resource).

Reply:
The first four suggested tests now exist.

Testing for file deletion on rename may be possible if the test can directly interact with the S3 mock, doing HEAD requests to check whether each file exists.

Reply:
The file deletion testing wasn't working - Moto was reporting 404 whenever I tried to access it directly, not sure why - so I've dropped those for now.